instrumentation/async-http-client-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/ResponseAdvice.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.asynchttpclient;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.instrumentation.api.ContextStore;
+import io.opentelemetry.javaagent.instrumentation.api.InstrumentationContext;
+import io.opentelemetry.javaagent.instrumentation.api.Pair;
+import net.bytebuddy.asm.Advice;
+import org.asynchttpclient.AsyncCompletionHandler;
+import org.asynchttpclient.AsyncHandler;
+import org.asynchttpclient.Response;
+
+public class ResponseAdvice {
+
+  @Advice.OnMethodEnter(suppress = Throwable.class)
+  public static Scope onEnter(
+      @Advice.This AsyncCompletionHandler<?> handler, @Advice.Argument(0) Response response) {
+
+    // TODO I think all this should happen on exit, not on enter.
+    // After response was handled by user provided handler.
```

Comment:
I thought you only thought this for sync handlers?

Reply:
Removed this comment

Reply:
We talked about this recently in some other PR. About in what context response processing should take place. You convinced me :) I will remove this comment from AHC 1.9 as well.